### ISSUE TITLE
Announcement: assignment_01 deadline EXTENDED to Saturday March 7th!

### ISSUE THREAD

**Issue by firasm** [STAT547-UBC-2019-20/Discussions#6]
Hello @STAT547-UBC-2019-20/students,

In case you didn't hear in class today or weren't there, due to the git workflow stuff taking longer than expected (see #3) I have decided to extend the deadline of Assignment 1 to NEXT Saturday, March 7th, 2020. 

Please note that:

- Milestone 1 is **still** due this Saturday!
- To keep the pace, both assignment 1 and 2 will be due on Saturday!

Let us know if you have any questions - and just as a sanity check for me, it's helpful if you could just give a thumbs up to announcements so I know people have seen them :-)

See you next week! Have fun with your projects!

**Comment by hanrach** [comment]
Is there a participation portion to be handed in for the git workflow lecture?

**Comment by firasm** [comment]
Yes, please fill in this part only :

The rest of the participation worksheet, we will do next Tuesday for cm103

> 16.2.7 Final: Provide links for the various sections of the git activity:
> 
> Link for your fork of the project repo: ## YOUR LINK HERE
> Link to the Pull request (PR) you sent in the main repo: ## YOUR LINK HERE
> Link to the commit after fixing the merge conflict in the main repo : ## YOUR LINK HERE
> Link to the commit after you pulled changes from the upstream branch: ## YOUR LINK HERE
> Happy Git with R is the most relevant for new R users, it was written by Jenny Bryan and others.

**Comment by hadleyd2** [comment]
Hello @firasm,

Can you possibly help me with this participation activity?  I have linked my fork repo as per the first activity: https://github.com/hadleyd2/group_09.git

I had created a pull request while in class and got a merge conflict. I had to resolve the merge conflict manually by deleting the conflict, and then I was able to commit the changes. There is no record of the merge conflict in my pull request or commit history, only the successful commit. How would I link to the merge conflict? Do I create a pull request with a merge conflict and just leave it open?

I am having some trouble following which commits are relating to what. For example, here is a link to the last commit on my main project repo:https://github.com/STAT547-UBC-2019-20/group_09/commit/b595eedc4fc9617ccb6551748e8ab3f05188c546
If you look at the "title" (for lack of a better term) it reads "Merge pull request #5 from hadleyd2/dan3"

Now, here is the associated commit in the fork of the main project repo, https://github.com/hadleyd2/group_09/commit/b595eedc4fc9617ccb6551748e8ab3f05188c546
This has a "title" of "Merge pull request STAT547-UBC-2019-20#5 from hadleyd2/dan3". 

Do you know which one refers to me updating the main repo from my branch and which one refers to my updating my fork from the main repo? How am I supposed to read these commits? I am having trouble interpreting these commits.

Thanks!
-Dan

**Comment by firasm** [comment]
hey @hadleyd2 I'm just going to put together a quick primer so things are written down in a more formal way. 

To answer your questions:

1. You're right, if you have a merge conflict you cannot actually commit the conflict to your repo. You need to resolve the merge conflict (as you did), and then commit the "fixed" version. In the comments, it's usually good practice to indicate that there was a merge conflict and that you resolved it (for archiving purposes). 

The line item is actually: 

> Link to the commit **after** fixing the merge conflict (emphasis added).

So the idea was to intentionally cause a merge conflict by editing the same file, resolve it, push the fixed version, and then link to this commit where you fixed it.

So in short, I think all you need to do is link to the commit after the one where you fixed the merge conflict. In the future, I encourage you to add a quick phrase like "fixed merge conflict in file XX". 

More to come in the git primer, [section 20 of HappygitwithR](https://happygitwithr.com/git-basics.html) might be useful for terminology

**Comment by dy-lin** [comment]
Hi @firasm, do the 48h grace periods apply to both assignments and milestones?

**Comment by firasm** [comment]
Yes 👍🏽

**Comment by firasm** [comment]
Yes 👍🏽
